Rudder-Moengage/Classes/RudderMoengageIntegration.m

```diff
@@ -66,83 +81,73 @@ - (void) processRudderEvent: (nonnull RSMessage *) message {
-    }
     if (traits != nil) {
       //set all predefined fields
       if (userId != nil) {
-        [[MoEngage sharedInstance] setUserUniqueID:userId];
-        [traits removeObjectForKey:@"id"];
+        [[MoEngageSDKAnalytics sharedInstance] setUniqueID:userId];
       }
-
+      
       if ([traits objectForKey:@"email"]) {
-        [[MoEngage sharedInstance] setUserEmailID:[traits objectForKey:@"email"]];
-        [traits removeObjectForKey:@"email"];
+        [[MoEngageSDKAnalytics sharedInstance] setEmailID:[traits objectForKey:@"email"]];
       }
-
+      
       if ([traits objectForKey:@"name"]) {
-        [[MoEngage sharedInstance] setUserName:[traits objectForKey:@"name"]];
-        [traits removeObjectForKey:@"name"];
+        [[MoEngageSDKAnalytics sharedInstance] setName:[traits objectForKey:@"name"]];
       }
-
+      
       if ([traits objectForKey:@"phone"]) {
-        [[MoEngage sharedInstance] setUserMobileNo:[traits objectForKey:@"phone"]];
-        [traits removeObjectForKey:@"phone"];
+        [[MoEngageSDKAnalytics sharedInstance] setMobileNumber:[traits objectForKey:@"phone"]];
       }
-
+      
       if ([traits objectForKey:@"firstName"]) {
-        [[MoEngage sharedInstance] setUserAttribute:[traits objectForKey:@"firstName"] forKey:USER_ATTRIBUTE_USER_FIRST_NAME];
-        [traits removeObjectForKey:@"firstName"];
+        [[MoEngageSDKAnalytics sharedInstance] setFirstName:[traits objectForKey:@"firstName"]];
+      } else if ([traits objectForKey:@"firstname"]) {
+        [[MoEngageSDKAnalytics sharedInstance] setFirstName:[traits objectForKey:@"firstname"]];
       }
-
+      
       if ([traits objectForKey:@"lastName"]) {
-        [[MoEngage sharedInstance] setUserLastName:[traits objectForKey:@"lastName"]];
-        [traits removeObjectForKey:@"lastName"];
-      }
-        
-      if ([traits objectForKey:@"firstname"]) {
-        [[MoEngage sharedInstance] setUserAttribute:[traits objectForKey:@"firstname"] forKey:USER_ATTRIBUTE_USER_FIRST_NAME];
-        [traits removeObjectForKey:@"firstname"];
+        [[MoEngageSDKAnalytics sharedInstance] setLastName:[traits objectForKey:@"lastName"]];
+      } else if ([traits objectForKey:@"lastname"]) {
+        [[MoEngageSDKAnalytics sharedInstance] setLastName:[traits objectForKey:@"lastname"]];
       }
-
-      if ([traits objectForKey:@"lastname"]) {
-        [[MoEngage sharedInstance] setUserLastName:[traits objectForKey:@"lastname"]];
-        [traits removeObjectForKey:@"lastname"];
-      }
-        
-      if ([traits objectForKey:@"gender"]) {
-        [[MoEngage sharedInstance] setUserAttribute:[traits objectForKey:@"gender"] forKey:USER_ATTRIBUTE_USER_GENDER];
-        [traits removeObjectForKey:@"gender"];
+      
+      NSString *gender =[traits objectForKey:@"gender"];
+      if (gender) {
+        if ([gender  isEqual: @"M"] || [gender  isEqualToString: @"MALE"]) {
```

Comment:
Can we use only 1 of them by our specs?

Reply:
We don't seems to have any specs defined w.r.t. Gender.
Only instance of gender reference, I could find in the Facebook Custom Audience doc, refer [here](https://www.rudderstack.com/docs/destinations/streaming-destinations/fb-custom-audience/#schema-fields-mapping:~:text=Use%20these%20values,for%20female.)

Reply:
Moreover, I tried to sync the gender code as per MoEngage Android implementation.

Reply:
Cool.